content/3_Silver/Prefix_Sums.mdx

```diff
@@ -681,6 +692,299 @@ as expected.
+		<tr>
+			<td>1</td>
+			<td>0</td>
+			<td>1</td>
+			<td>1</td>
+		</tr>
+		<tr>
+			<td>1</td>
+			<td>1</td>
+			<td>0</td>
+			<td>0</td>
+		</tr>
+		<tr>
+			<td>1</td>
+			<td>1</td>
+			<td>1</td>
+			<td>1</td>
+		</tr>
+	</table>
+</center>
+
+Then, we apply the preprocessing step.
+
+$$
+\sum_{i=1}^{N} \sum_{j=1}^{N} \texttt{prefix}[i][j] = \texttt{prefix}[i][j] + \texttt{prefix}[i-1][j] + \texttt{prefix}[i][j-1] - \texttt{prefix}[i-1][j-1]
+$$
+
+For our example, we get 
+
+<center>
+	<table className='text-center'>
+	<tr>
+		<td>0</td>
+		<td>0</td>
+		<td>0</td>
+		<td>0</td>
+		<td>0</td>
+	</tr>
+	<tr>
+		<td>0</td>
+		<td>0</td>
+		<td>1</td>
+		<td>1</td>
+		<td>1</td>
+	</tr>
+	<tr>
+		<td>0</td>
+		<td>1</td>
+		<td>2</td>
+		<td>3</td>
+		<td>4</td>
+	</tr>
+	<tr>
+		<td>0</td>
+		<td>2</td>
+		<td>4</td>
+		<td>5</td>
+		<td>6</td>
+	</tr>
+	<tr>
+		<td>0</td>
+		<td>3</td>
+		<td>6</td>
+		<td>8</td>
+		<td>10</td>
+	</tr>
+	</table>
+</center>
+
+Notice how all queries can now be processed using 
+
+$$
+\begin{aligned}
+\texttt{arr}[i][j]=&\,\texttt{prefix}[A][B]
+        - \texttt{prefix}[a-1][B] \\
+        &- \texttt{prefix}[A][b-1] + \texttt{prefix}[a-1][b-1]
+\end{aligned}
+$$
+
+#### Forest Queries Solution
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+#define vt vector
+
+#define FOR(i, a, b) for(int i = (a); i < (b); i++)
+#define FORE(i, a, b) for(int i = (a); i <= (b); i++)
+#define F0R(i, a) for(int i = 0; i < (a); i++)
+#define trav(a, x) for (auto& a : x)
+
+void setIn(string s) { freopen(s.c_str(),"r",stdin); }
+void setOut(string s) { freopen(s.c_str(),"w",stdout); }
+void setIO(string s = "") {
+    ios_base::sync_with_stdio(0); cin.tie(0); cout.tie(0);
+    if (!s.empty()) { setIn(s+".in"), setOut(s+".out"); }
+}
+
+int N, Q;
+int pfx[1001][1001];
+
+int main() {
+    ios_base::sync_with_stdio(0);
+    cin.tie(0);
+
+    cin >> N >> Q;
+    F0R(i, N) {
+        F0R(j, N) {
+            char a; cin >> a;
+            if(a == '*') pfx[i+1][j+1]++;
+        }
+    }
+
+    FOR(i, 1, N+1) {
+        FOR(j, 1, N+1){ 
+            pfx[i][j] += pfx[i-1][j] + pfx[i][j-1] - pfx[i-1][j-1];
+        }
+    }
+
+    F0R(i, Q) {
+        int x1, x2, y1, y2;
+        cin >> x1 >> y1 >> x2 >> y2;
+        cout << pfx[x2][y2] - pfx[x1-1][y2] - pfx[x2][y1-1] + pfx[x1-1][y1-1] << "\n";
+    }
+}
+```
+</CPPSection>
+
+</LanguageSection>
+
+<Warning>
+
+We need to be cautious of off-by-one errors, as intervals can be inclusive, exclusive, 1-indexed, etc.
+
+</Warning>
+
+#### Rectangle
+
+Given a rectangle with endpoints $(x1, y1), (x2, y2)$, we can compute the "layers" of many such rectangles bounded together through the use of increment and decrement points. Since our endpoints are not inclusive, our increment points are $(x1, y1),(x2, y2)$, and decrement points are $(x1, y2), (x2, y1)$. Our prefix precomputation remains the same as described in the previous section.
```

Comment:
> Since our endpoints are not inclusive, our increment points are $(x1, y1),(x2, y2)$, and decrement points are $(x1, y2), (x2, y1)$.

Make sure to mention the reason.